### DISCUSSION TITLE
Subgraph Embedding

### DISCUSSION THREAD

**Discussion by elbasir** [pyg-team/pytorch_geometric#4049] [Q&A]
Hi,
Assume I have a graph G of 5000 nodes indirectly connected and I have several subgraphs within G that I am interested in to retrieve an embedding for each subgraph to be used for some downstream tasks. Can GNN or GAE handle this? If yes, can you please refer me to such an example. I am new to the field and I'd like to learn more about it.

Best,

**Comment by rusty1s** [comment]
If you know about the assignment of nodes to clusters/subgraphs, you can utilize `torch_scatter` to aggregate node embeddings to subgraph-level embeddings. Let me know if that answers your question :)

**Reply by elbasir** [reply]
Hi again,
Yes, I have nodes that belong to each subgraph. Do you mean here, aggregating node embeddings belong to each subgraph?

**Reply by rusty1s** [reply]
Yes, that's what you are trying to achieve, right?

Given that you have a node-level vector `cluster`, in which each entry denotes the assignment of the node to one of `0` to `C-1` clusters, then:
```python
from torch_scatter import scatter

x_subgraph = scatter(x, cluster, dim=0, reduce="mean")
```
will give you the aggregated subgraph embeddings of shape `[num_subgraphs, num_features]`.

**Reply by elbasir** [reply]
Yes, I want to retrieve a subgraph embedding. Thanks for your answer. Just to clarify, embedding of each node will be generated from the entire graph then applying torch_scatter to aggregate node embeddings to subgraph-level embeddings!

Does "x" here represent the node embedding and "cluster" represent the subgraph that node embedding "x" belong to?

**Reply by rusty1s** [reply]
Yes, `x` is of shape `[num_nodes, num_features]`, and `cluster` is a node-level vector containing indices from `0` to `C - 1` where `C` denotes the number of subgraphs in your graph.

**Comment by elbasir** [comment]
Thanks. I appreciate your time.
